### PR TITLE
#15399 Supports Babelfish session manager

### PR DESCRIPTION
**Pull request by LonwoLonwo**
2022-02-07 10_48_46-DBeaver 21 3 4 - Sessions - demo2

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/session/BabelfishSession.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+    @Override
+    public String getActiveQuery()
+    {
+        return query;
+    }
+
+    @Override
+    public String toString()
+    {
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/session/BabelfishSession.java:147]: Please fix formatting.

File: plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/session/BabelfishSession.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        if (query != null && query.length() > 50) {
+            return CommonUtils.truncateString(query, 50) + " ...";
+        } else {
+            return query;
+        }
+    }
+
+    @Override
+    public String getActiveQuery()
+    {
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/session/BabelfishSession.java:141]: Please fix formatting.

File: plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/session/BabelfishSession.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    }
+
+    @Property(category = CAT_TIMING, order = 33)
+    public Date getStateChange() {
+        return stateChange;
+    }
+
+    @Property(viewable = true, order = 7)
+    public String getState()
+    {
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/session/BabelfishSession.java:126]: Please fix formatting.

File: plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/session/BabelfishSession.java [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    }
+
+    @Property(category = CAT_CLIENT, order = 4)
+    public String getClientPort() {
+        return clientPort;
+    }
+
+    @Property(viewable = true, order = 5)
+    public String getDb()
+    {
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/session/BabelfishSession.java:95]: Please fix formatting.

File: plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/session/BabelfishSession.java [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+
+    @Property(viewable = true, category = CAT_CLIENT, order = 2)
+    public String getUser()
+    {
+        return user;
+    }
+
+    @Property(category = CAT_CLIENT, order = 3)
+    public String getClientHost()
+    {
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/session/BabelfishSession.java:84]: Please fix formatting.

File: plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/session/BabelfishSession.java [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+        this.queryStart = JDBCUtils.safeGetTimestamp(dbResult, "query_start");
+        this.stateChange = JDBCUtils.safeGetTimestamp(dbResult, "state_change");
+
+        this.state = JDBCUtils.safeGetString(dbResult, "state");
+        this.appName = JDBCUtils.safeGetString(dbResult, "application_name");
+    }
+
+    @Property(viewable = true, order = 1)
+    public int getPid()
+    {
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/model/session/BabelfishSession.java:72]: Please fix formatting.

**Comment by serge-rider** [comment]
We need new extension for Babelfish. It may depends on both SQL Server and PostgreSQL bundles.

**Pull request closed without merging**